### PR TITLE
feat(dialogs): add explicit dialog option translation key support; add missing dialog start arguments

### LINKED ISSUE #796: ASHES Doesn't Fully Implement Translation Keys for Dialog Options
**Describe the bug**
ASHES needs to finish implementing translation keys for dialog options, including the avatar/default/timeout options.

**Versions**
* Escoria: 4.0.0-alpha.272
* Godot: 4.4

### LINKED ISSUE #797: ASHES Needs to Finish Implementing Dialog Options
**Describe the bug**
The new ASHES scripting language needs to finish implementing the options that ESCScript used for dialogs, namely:

- Avatar path
- Timeout (in seconds)
- Timeout option

Because all are optional, but the timeout-related ones are dependent on each other, discussion is needed on how best to represent the arguments in ASHES.

(Recall that ASHES uses a `?!` on its own line to indicate a dialog block.)

Also, because these options have never been placed into a test room, this will need to be addressed as well to help avoid regressions and demonstrate how to make use of these features.)

**To Reproduce**

**Expected behavior**
- If an avatar path is specified, the corresponding avatar should be shown by the dialog UI.
- If timeout and timeout option are specified, the dialog in question should respect the timeout value, after which the dialog should auto-select the timeout option specified.

**Screenshots**

**Versions**
* Escoria: 4.0.0-alpha.272
* Godot: 4.4

**Additional context**

### PR DESCRIPTION
**Pull request by BHSDuncan**
Closes #796 

## Summary by CodeRabbit

* **New Features**
  * Dialog options now support explicit translation keys as a primary localization mechanism
  * Dialog start statements accept optional runtime arguments for avatar, timeout, and timeout_option settings
  * Implemented validation for dialog start arguments with proper error handling

* **Tests**
  * Added comprehensive test coverage for dialog translation keys and start argument propagation, including edge cases for invalid inputs

### CODE REVIEW

File: addons/escoria-core/game/core-scripts/esc/compiler/esc_interpreter.gd [hunk 1]
```diff
@@ -473,16 +473,72 @@ func visit_global_stmt(stmt: ESCGrammarStmts.Global):
 func visit_dialog_stmt(stmt: ESCGrammarStmts.Dialog):
 	var dialog: ESCDialog = ESCDialog.new()
 	var rc = ESCExecution.RC_OK
+	var dialog_args := stmt.get_args()
 
 	_dialog_depth += 1
 
+	if dialog_args.size() > 0:
+		var avatar = await _evaluate(dialog_args[0])
```
Comment (StraToN) [inline addons/escoria-core/game/core-scripts/esc/compiler/esc_interpreter.gd:481]: Isn't `avatar` a specific use case of `simple-dialogs` ? I fear using avatar mention in core script may create a hard relation with simple-dialogs.

**Review by StraToN — APPROVED** [review]

File: addons/escoria-core/game/core-scripts/esc/compiler/esc_interpreter.gd (hunk 1 above)
Comment (BHSDuncan) [reply addons/escoria-core/game/core-scripts/esc/compiler/esc_interpreter.gd:481]: As a follow-up from our conversation on Discord: `avatar` is included in the `ESCDialog` wrapper and is passed into the configured dialog player.

**Comment by BHSDuncan** [comment]
Also addresses #797
